vault/hcp_link/capabilities/meta/meta.go

```diff
@@ -189,3 +191,110 @@ func (h *hcpLinkMetaHandler) ListAuths(ctx context.Context, req *meta.ListAuthsR
 		Auths: auths,
 	}, nil
 }
+
+func (h *hcpLinkMetaHandler) GetClusterStatus(ctx context.Context, req *meta.GetClusterStatusRequest) (*meta.GetClusterStatusResponse, error) {
+	if h.wrappedCore.HAState() != consts.Active {
```

Comment:
Wondering if we would need to guard accessing `hcpLinkMetaHandler` members with a lock? What do you think?

Reply:
I guess I am wrong with accessing `wrappedCore`. I was looking at the start and stop functions and see how we needed to use a lock. I understand that, and here it seems safe to access `wrappedCore` as is. We are just reading stuff, so not really a big concern I guess.

Reply:
I think we need to think about using the `HAStateI()`. This function accesses some core members such as `c.perfStandby` without a guard and using it here means that we are not going through our regular request handling path during which a state lock is grabbed. I checked some usages of that function, and it seems that some sort of lock was held prior to calling that function. So, I am wondering if we should introduce a function that uses a guard? What do you think?

Reply:
A bit of an update: I did a bit of digging regarding the methods added or used for the `GetClusterStatus` handler to determine what our locking strategy should be. The TL;DR is that I think we're mostly fine but I'll summarize here:

- `HAEnabled` : `c.ha` is only set during initialization of a new core, so this only then exposes the underlying call to `c.ha.HAEnabled`
- `HAState`: As we discussed this should have a lock, I will add a function `HAStateWithLock` that grabs `c.stateLock.RLock`
- `GetHAPeerNodesCached`: This uses `c.clusterPeerClusterAddrsCache` which is a `*cache.Cache` that has locking built in. For example, calling `c.clusterPeerClusterAddrsCache.Items()` uses an underlying lock built into the cache
- `GetRaftConfiguration`: This uses `c.ha` mainly to cast to a `RaftBackend` which mentioned above should be safe. Then the underlying call to `RaftBackend.GetConfiguration` is guarded by the `RaftBackend.l`
- `GetRaftAutopilotState`: This uses `c.ha` mainly to cast to a `RaftBackend` which mentioned above should be safe. Then the underlying call to `RaftBackend.GetAutopilotServerState` is guarded by `RaftBackend.l`
- `StorageType`: The storage type shouldn't change
- `ClusterID`: The cluster ID won't be changing and we call this elsewhere without locking so should be safe